### DISCUSSION TITLE
Alternative audio decoders

### DISCUSSION THREAD

**Discussion by katspaugh** [katspaugh/wavesurfer.js#2702] [Ideas]
WebAudio decoder is very memory intensive and cannot easily be done in chunks (there’s a [library](https://github.com/soundcut/decode-audio-data-fast) that does it but only for mp3s).

We could explore and profile alternatives like this wasm-based set of decoders: https://github.com/eshaz/wasm-audio-decoders

**Comment by prakharpbuf** [comment]
We could create our own wasm port of ffmpeg or use one of the existing ones:
https://github.com/ffmpegwasm/ffmpeg.wasm
https://github.com/Kagami/ffmpeg.js

**Reply by katspaugh** [reply]
Nice! I didn't know someone already ported it.

**Reply by sundayz** [reply]
I'm pretty sure that emscripten ports of ffmpeg are too big to realistically use in a browser app, unless you want to include a 10MB dependency on top of everything else in your app. I don't know about the wasm version, but I assume it's also pretty big.

**Reply by prakharpbuf** [reply]
it is pretty big indeed

**Reply by katspaugh** [reply]
In the project I linked, each decoder seems to be 60-70 Kb.
Of course, ffmpeg has the advantage of supporting all possible formats.

**Reply by sundayz** [reply]
It would be neat to have a plugin or maybe a backend (when we implement an `AudioBackend` interface) that can use ffmpeg, but it would have to be optional and loaded on-demand rather than part of the wavesurferjs bundle.

**Reply by thijstriemstra** [reply]
> We could create our own wasm port of ffmpeg

my gosh, you have any idea how much work that is?
